CHANGES.txt

```diff
@@ -12,6 +12,12 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
   From William Deegan:
     - Fix reproducible builds. Restore logic respecting SOURCE_DATE_EPOCH when set.
     - Fix version tests to work with updated scons --version output. (Date format changed)
+    
+  From Mats Wichmann:
+    - Unify the NodeList definitions - there were two, and didn't need to be.
+      Two unit tests had to be updated to make sure they're expecting the
+      right string representation. Alias now returns a NodeList like other
```

Comment:
Probably should explain the change to str(NodeList) here in case someone is depending on it.
Also.. they could cause unnecessary rebuilds on upgrade if the str(NodeList) is used anywhere generating a sig.
Also should list all the methods/apis which now return a NodeList() when before they returned a list `[]`.
Lastly.. any idea if this has any notable perf degradation?

Reply:
Would have to write somethng synthetic to check the performance, nothing we have in the testsuite takes enough time in NodeLists to show any significant difference.

I'll queue up trying to list the changes.  In general following the "builders return a NodeList" rule, even for things which aren't classically builders (Builder classes) but are documented as builders.

Reply:
as a first step, I've added some information in the PR description above (see Edited section).